### PR TITLE
Brice/support multiple yamls

### PR DESCRIPTION
**Pull request by bricerisingalgorand**
With this we'll be able to support multiple mule yamls and deep merge them in the order that they were passed in:

for example:
```
mule -f test/test-yamls/valid/overrides-test.yaml -f test/test-yamls/valid/overrides-test.yaml echo-configs-override
```

### CODE REVIEW

File: mule/driver.py [hunk 1]
```diff
@@ -29,6 +32,16 @@ def main():
...
 
+def _read_mule_yamls(mule_yamls):
+    mule_config = {}
+    for mule_yaml in mule_yamls:
+        update_dict(
+            mule_config,
+            file_util.readYamlFile(mule_yaml),
+            overwrite_lists=False
+        )
+    return mule_config
```
Comment (egieseke) [inline mule/driver.py:43]: Is it possible that there could be conflicts in the mule.yaml files?  Should there be a warning if conflicts are found?

**Review by egieseke — APPROVED** [review]
Looks good.

File: mule/parser.py [hunk 2]
```diff
@@ -16,7 +16,8 @@ def parseArgs():
     parser.add_argument(
         '-f',
         '--file',
-        default = 'mule.yaml',
+        action='append',
+        default = [],
```
Comment (btoll) [inline mule/parser.py:20]: What about:
```
default = ['mule.yaml']
```

File: mule/driver.py (hunk 1 above)
Comment (bricerisingalgorand) [reply mule/driver.py:43]: It's implemented such that subsequent files overwrite values in previous files, except when both values are lists the contents are appended

File: mule/parser.py (hunk 2 above)
Comment (bricerisingalgorand) [reply mule/parser.py:20]: If we have this here as default, it'll be included in the list when files are passed in. I'm not sure that's what people expect when they provide one or more yaml files to the cli

for example, this command would produce this list
```
mule -f test/mule.yaml test

[ "mule.yaml", "test/mule.yaml" ]
```

File: mule/driver.py [hunk 3]
```diff
@@ -11,7 +11,10 @@
 def main():
     args = mule.parser.parseArgs()
     try:
-        mule_config = file_util.readYamlFile(args.file)
+        mule_yamls = args.file
+        if len(args.file) == 0:
+            mule_yamls.append('mule.yaml')
```
Comment (btoll) [inline mule/driver.py:16]: Could be avoided if setting the default like I mentioned below.

File: mule/parser.py (hunk 2 above)
Comment (btoll) [reply mule/parser.py:20]: Makes sense.

File: mule/driver.py (hunk 3 above)
Comment (btoll) [reply mule/driver.py:16]: This is moot in light of what you wrote to first comment.

**Review by btoll — APPROVED** [review]

**Review by onetechnical — APPROVED** [review]

**Pull request merged**